In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import random
random.seed(654)

Import the data set.

In [2]:
bbq = pd.read_csv('bbq_summer.csv')
bbq.head()

bundleID                starter    maindishI maindishII  \
0         1  Fried Chicken Tenders  BBQ Brisket    Sausage   
1         2  Fried Chicken Tenders  BBQ Brisket    Sausage   
2         3  Fried Chicken Tenders  BBQ Brisket    Sausage   
3         4  Fried Chicken Tenders  BBQ Brisket    Sausage   
4         5  Fried Chicken Tenders  BBQ Brisket    Sausage   

                 side              dessert  avg_rating  
0      Mac and Cheese      Blondie Brownie        6.04  
1      Mac and Cheese  Apple Pie a la Mode        9.29  
2        Potato Salad      Blondie Brownie        6.20  
3        Potato Salad  Apple Pie a la Mode        8.84  
4  French Fry Platter      Blondie Brownie        8.24

Remove the index.

In [3]:
bbq = bbq.drop(columns='bundleID')
bbq.head()

starter    maindishI maindishII                side  \
0  Fried Chicken Tenders  BBQ Brisket    Sausage      Mac and Cheese   
1  Fried Chicken Tenders  BBQ Brisket    Sausage      Mac and Cheese   
2  Fried Chicken Tenders  BBQ Brisket    Sausage        Potato Salad   
3  Fried Chicken Tenders  BBQ Brisket    Sausage        Potato Salad   
4  Fried Chicken Tenders  BBQ Brisket    Sausage  French Fry Platter   

               dessert  avg_rating  
0      Blondie Brownie        6.04  
1  Apple Pie a la Mode        9.29  
2      Blondie Brownie        6.20  
3  Apple Pie a la Mode        8.84  
4      Blondie Brownie        8.24

In [4]:
bbq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 384 entries, 0 to 383
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   starter     384 non-null    object 
 1   maindishI   384 non-null    object 
 2   maindishII  384 non-null    object 
 3   side        384 non-null    object 
 4   dessert     384 non-null    object 
 5   avg_rating  384 non-null    float64
dtypes: float64(1), object(5)
memory usage: 18.1+ KB


Besides "avg_rating", all variables are categorical.

In [5]:
bbq.describe()

avg_rating
count  384.000000
mean     7.291298
std      1.740623
min      1.690000
25%      6.065000
50%      7.520000
75%      8.746000
max      9.940000

The mean value for average rating is 7.29, with a standard deviation of 1.74. Since the mean is lower than the median, the average rating is left-skewed. 

Dummify the categorical variables.

In [6]:
bbq = pd.get_dummies(bbq, drop_first=True, columns=['starter','maindishI','maindishII','side','dessert'])
bbq.head()

avg_rating  starter_Fried Chicken Tenders  starter_Jumbo Shrimp Cocktail  \
0        6.04                              1                              0   
1        9.29                              1                              0   
2        6.20                              1                              0   
3        8.84                              1                              0   
4        8.24                              1                              0   

   starter_Sticky Chicken Tenders  maindishI_BBQ Chicken  \
0                               0                      0   
1                               0                      0   
2                               0                      0   
3                               0                      0   
4                               0                      0   

   maindishI_Buffalo Chicken Macaroni Stacker  maindishI_Pulled Pork  \
0                                           0                      0   
1                                           0                      0   
2                                           0                      0   
3                                           0                      0   
4                                           0                      0   

   maindishII_Lobster Roll  maindishII_Sausage  maindishII_Steak Sampler  \
0                        0                   1                         0   
1                        0                   1                         0   
2                        0                   1                         0   
3                        0                   1                         0   
4                        0                   1                         0   

   side_Mac and Cheese  side_Potato Salad  dessert_Blondie Brownie  
0                    1                  0                        1  
1                    1                  0                        0  
2                    0                  1                        1  
3                    0                  1                        0  
4                    0                  0                        1

Build a linear model.

In [7]:
bbq.columns

Index(['avg_rating', 'starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie'],
      dtype='object')

In [8]:
X = bbq[['starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie']]
y = bbq['avg_rating']

In [9]:
from sklearn.linear_model import LinearRegression
from sklearn import metrics

regressor = LinearRegression()
regressor.fit(X,y)

LinearRegression()

Display the coefficient values.

In [10]:
coef_df = pd.DataFrame(regressor.coef_, X.columns, columns=['Coefficients'])
coef_df

Coefficients
starter_Fried Chicken Tenders                   0.133571
starter_Jumbo Shrimp Cocktail                  -0.630587
starter_Sticky Chicken Tenders                  0.339275
maindishI_BBQ Chicken                          -0.209208
maindishI_Buffalo Chicken Macaroni Stacker      0.536821
maindishI_Pulled Pork                           0.529313
maindishII_Lobster Roll                         1.048096
maindishII_Sausage                              0.331992
maindishII_Steak Sampler                       -0.500463
side_Mac and Cheese                             0.575125
side_Potato Salad                              -0.022194
dessert_Blondie Brownie                        -0.856204

Optimal bundle:

starter: Sticky Chicken Tenders

maindish I: Buffalo Chicken Macaroni Stacker

maindish II: Lobster Roll

side: Mac and Cheese

dessert: Apple Pie a la Mode

In [11]:
vendor_costs = pd.read_csv('vendor_costs_spring_2022 - Sheet1.csv')
vendor_costs

Item Item Category  Cost Per Serving (Dollars)
0              Fried Chicken Tenders      Starters                        3.40
1                        Bacon Fries      Starters                        3.10
2             Sticky Chicken Tenders      Starters                        2.90
3              Jumbo Shrimp Cocktail      Starters                        4.60
4                        BBQ Brisket   Main Dish I                        6.00
5                        Pulled Pork   Main Dish I                        5.70
6                        BBQ Chicken   Main Dish I                        5.30
7   Buffalo Chicken Macaroni Stacker   Main Dish I                        6.10
8                            Sausage  Main Dish II                        4.70
9                       Lobster Roll  Main Dish II                        5.20
10                   Beef Short Ribs  Main Dish II                        6.00
11                     Steak Sampler  Main Dish II                        4.70
12                    Mac and Cheese          Side                        0.45
13                      Potato Salad          Side                        0.35
14                French Fry Platter          Side                        0.15
15                   Blondie Brownie       Dessert                        0.90
16               Apple Pie a la Mode       Dessert                        1.10

In [12]:
2.90 + 6.10 + 5.20 + 0.45 + 1.10

15.749999999999998

The cost for the optimal bundle is $15.75.

In [13]:
2.90 + 5.70 + 5.20 + 0.45 + 1.10

15.35

We decided to replace the first main dish with Pulled Port since it has similar influence on ratings but has a lower cost compared to Buffalo Chicken Macaroni Stacker.

So the new bundle will be:

starter: Sticky Chicken Tenders

maindish I: Pulled Pork

maindish II: Lobster Roll

side: Mac and Cheese

dessert: Apple Pie a la Mode

The cost for the bundle in which we use Pulled Pork to replace Buffalo Chicken Macaroni Stacker is $15.35.

In [14]:
15.99 - (2.90 + 5.70 + 5.20 + 0.45 + 1.10)

0.6400000000000006

The profit margin for the standard bundle is $0.64.

Create a kid's meal with only the starter (Sticky Chicken Tenders), a main dish (Lobster Roll), and the dessert (Apple Pie a la Mode). 

In [15]:
2.90 + 5.20 + 1.10

9.2

The cost for the kid's meal is $9.2.

In [16]:
9.99 - (2.90 + 5.20 + 1.10)

0.7900000000000009

The profit margin for the kid's meal is $0.79.

### Summer Grill Menu Strategies

Lobster Land should position the Summer Grill Menu as a propagandistic event. It is unnecessary for the park to pursue large interests in this event. Still, Lobster Land should apply a pricing strategy that can fulfill the need for variable and setup cost recovery. 

The rating-based conjoint analysis suggests that selecting Sticky Chicken Tenders, Buffalo Chicken Macaroni Stacker, Lobster Roll, Mac and Cheese, and Apple Pie a la will best satisfy the visitors. This bundle incurs a cost of 15.75 dollars. Since Pulled Pork and Buffalo Chicken Macaroni Stacker almost contribute to the rating equally while Pulled Pork has a lower unit cost, it will be better for Lobster Land to replace the first main dish with Pulled Pork. Hence, the recommended options for Lobster Land to put in the Summer Grill Menu are Sticky Chicken Tenders, Pulled Pork, Lobster Roll, Mac and Cheese, and Apple Pie a la. The cost per bundle is 15.35 dollars.

Lobster Land has always been an enjoyable spot for families. Therefore, besides the standard bundle, Lobster Land should apply a versioning strategy and push out a kid’s meal as well at a lower price with smaller serving size to serve the young visitors. The kid’s meal bundle should include Sticky Chicken Tenders as the starter, Lobster Roll as the main dish, and Apple Pie a la as the dessert. The cost per bundle is 9.2 dollars. 

To match up the design of the food bundles, Lobster Land should also adjust its pricing for this event. Because the cost of the standard meal exceeds 15 dollars, Lobster Land should raise the fee for the entrance of the Barbeque Tent. We suggest a Charm Pricing strategy by applying which Lobster Land can adjust the price of access to 15.99 dollars without discouraging the visitors from paying. For the kid’s meal, the unit price should be set similarly to 9.99 dollars. The profit margin is 0.64 dollars for the standard meal and 0.79 dollars for the kid’s meal. Lobster Land should also prepare enough penny coins to handle cash payment situations to get this strategy ready. 